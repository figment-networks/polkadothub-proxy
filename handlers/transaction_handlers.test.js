const transactionHandlers = require('./transaction_handlers');
const {createApi} = require('../tests/shared');

// These actually DO hit the API
describe('integration', () => {
  let call = {request: {height: 2197580}};
  let api;

  beforeAll(async (done) => {
    api = await createApi();
    done()
  });

  describe('getByHeight', () => {
    it('returns a result that matches snapshot', async () => {
      expect(await transactionHandlers.getByHeight(api, call)).toMatchSnapshot();
    });
  });
});
