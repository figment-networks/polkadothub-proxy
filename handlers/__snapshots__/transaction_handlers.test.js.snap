// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`integration getByHeight returns a result that matches snapshot 1`] = `
Object {
  "transactions": Array [
    Object {
      "args": "[{\\"callIndex\\":\\"0x0610\\",\\"args\\":{\\"era\\":671,\\"validators\\":[[\\"FG2vPisy1ocUrZupvZxXvuPmjqkWPabBv96h9atGAoud5B9\\",6],[\\"E8MhgpapcCqr5tnT5xmJnYX8Kuc85HaM4T2q23Cf3iqQCrX\\",6]]}}, {\\"callIndex\\":\\"0x0610\\",\\"args\\":{\\"era\\":672,\\"validators\\":[[\\"FG2vPisy1ocUrZupvZxXvuPmjqkWPabBv96h9atGAoud5B9\\",6],[\\"E8MhgpapcCqr5tnT5xmJnYX8Kuc85HaM4T2q23Cf3iqQCrX\\",6]]}}, {\\"callIndex\\":\\"0x0610\\",\\"args\\":{\\"era\\":673,\\"validators\\":[[\\"FG2vPisy1ocUrZupvZxXvuPmjqkWPabBv96h9atGAoud5B9\\",6],[\\"E8MhgpapcCqr5tnT5xmJnYX8Kuc85HaM4T2q23Cf3iqQCrX\\",6]]}}, {\\"callIndex\\":\\"0x0610\\",\\"args\\":{\\"era\\":674,\\"validators\\":[[\\"FG2vPisy1ocUrZupvZxXvuPmjqkWPabBv96h9atGAoud5B9\\",6],[\\"E8MhgpapcCqr5tnT5xmJnYX8Kuc85HaM4T2q23Cf3iqQCrX\\",6]]}}, {\\"callIndex\\":\\"0x0610\\",\\"args\\":{\\"era\\":675,\\"validators\\":[[\\"FG2vPisy1ocUrZupvZxXvuPmjqkWPabBv96h9atGAoud5B9\\",6],[\\"E8MhgpapcCqr5tnT5xmJnYX8Kuc85HaM4T2q23Cf3iqQCrX\\",6]]}}, {\\"callIndex\\":\\"0x0610\\",\\"args\\":{\\"era\\":676,\\"validators\\":[[\\"FG2vPisy1ocUrZupvZxXvuPmjqkWPabBv96h9atGAoud5B9\\",6],[\\"E8MhgpapcCqr5tnT5xmJnYX8Kuc85HaM4T2q23Cf3iqQCrX\\",6]]}}, {\\"callIndex\\":\\"0x0610\\",\\"args\\":{\\"era\\":677,\\"validators\\":[[\\"FG2vPisy1ocUrZupvZxXvuPmjqkWPabBv96h9atGAoud5B9\\",6],[\\"E8MhgpapcCqr5tnT5xmJnYX8Kuc85HaM4T2q23Cf3iqQCrX\\",6]]}}, {\\"callIndex\\":\\"0x0610\\",\\"args\\":{\\"era\\":678,\\"validators\\":[[\\"FG2vPisy1ocUrZupvZxXvuPmjqkWPabBv96h9atGAoud5B9\\",6],[\\"E8MhgpapcCqr5tnT5xmJnYX8Kuc85HaM4T2q23Cf3iqQCrX\\",6]]}}, {\\"callIndex\\":\\"0x0610\\",\\"args\\":{\\"era\\":679,\\"validators\\":[[\\"FG2vPisy1ocUrZupvZxXvuPmjqkWPabBv96h9atGAoud5B9\\",6],[\\"E8MhgpapcCqr5tnT5xmJnYX8Kuc85HaM4T2q23Cf3iqQCrX\\",6]]}}, {\\"callIndex\\":\\"0x0610\\",\\"args\\":{\\"era\\":680,\\"validators\\":[[\\"FG2vPisy1ocUrZupvZxXvuPmjqkWPabBv96h9atGAoud5B9\\",6],[\\"E8MhgpapcCqr5tnT5xmJnYX8Kuc85HaM4T2q23Cf3iqQCrX\\",6]]}}, {\\"callIndex\\":\\"0x0610\\",\\"args\\":{\\"era\\":681,\\"validators\\":[[\\"FG2vPisy1ocUrZupvZxXvuPmjqkWPabBv96h9atGAoud5B9\\",6],[\\"E8MhgpapcCqr5tnT5xmJnYX8Kuc85HaM4T2q23Cf3iqQCrX\\",6]]}}, {\\"callIndex\\":\\"0x0610\\",\\"args\\":{\\"era\\":682,\\"validators\\":[[\\"F3BPTmS1fExbuA3SnjNx4jLi5Ensc5hRnGNhY9onYw2ok2H\\",2],[\\"EXHRoKfGoCe414hrzYbG8SNCNVgBxSpu6rHUBkPNtZcj3om\\",1]]}}, {\\"callIndex\\":\\"0x0610\\",\\"args\\":{\\"era\\":683,\\"validators\\":[[\\"F3BPTmS1fExbuA3SnjNx4jLi5Ensc5hRnGNhY9onYw2ok2H\\",2]]}}, {\\"callIndex\\":\\"0x0610\\",\\"args\\":{\\"era\\":684,\\"validators\\":[[\\"F3BPTmS1fExbuA3SnjNx4jLi5Ensc5hRnGNhY9onYw2ok2H\\",2]]}}, {\\"callIndex\\":\\"0x0610\\",\\"args\\":{\\"era\\":685,\\"validators\\":[[\\"F3BPTmS1fExbuA3SnjNx4jLi5Ensc5hRnGNhY9onYw2ok2H\\",2]]}}, {\\"callIndex\\":\\"0x0610\\",\\"args\\":{\\"era\\":686,\\"validators\\":[[\\"F3BPTmS1fExbuA3SnjNx4jLi5Ensc5hRnGNhY9onYw2ok2H\\",2]]}}, {\\"callIndex\\":\\"0x0610\\",\\"args\\":{\\"era\\":687,\\"validators\\":[[\\"F3BPTmS1fExbuA3SnjNx4jLi5Ensc5hRnGNhY9onYw2ok2H\\",2]]}}, {\\"callIndex\\":\\"0x0610\\",\\"args\\":{\\"era\\":688,\\"validators\\":[[\\"F3BPTmS1fExbuA3SnjNx4jLi5Ensc5hRnGNhY9onYw2ok2H\\",2]]}}, {\\"callIndex\\":\\"0x0610\\",\\"args\\":{\\"era\\":689,\\"validators\\":[[\\"F3BPTmS1fExbuA3SnjNx4jLi5Ensc5hRnGNhY9onYw2ok2H\\",2]]}}, {\\"callIndex\\":\\"0x0610\\",\\"args\\":{\\"era\\":690,\\"validators\\":[[\\"F3BPTmS1fExbuA3SnjNx4jLi5Ensc5hRnGNhY9onYw2ok2H\\",2]]}}, {\\"callIndex\\":\\"0x0610\\",\\"args\\":{\\"era\\":691,\\"validators\\":[[\\"F3BPTmS1fExbuA3SnjNx4jLi5Ensc5hRnGNhY9onYw2ok2H\\",2]]}}, {\\"callIndex\\":\\"0x0610\\",\\"args\\":{\\"era\\":692,\\"validators\\":[[\\"F3BPTmS1fExbuA3SnjNx4jLi5Ensc5hRnGNhY9onYw2ok2H\\",2]]}}, {\\"callIndex\\":\\"0x0610\\",\\"args\\":{\\"era\\":693,\\"validators\\":[[\\"F3BPTmS1fExbuA3SnjNx4jLi5Ensc5hRnGNhY9onYw2ok2H\\",2]]}}, {\\"callIndex\\":\\"0x0610\\",\\"args\\":{\\"era\\":694,\\"validators\\":[[\\"F3BPTmS1fExbuA3SnjNx4jLi5Ensc5hRnGNhY9onYw2ok2H\\",2]]}}, {\\"callIndex\\":\\"0x0610\\",\\"args\\":{\\"era\\":695,\\"validators\\":[[\\"F3BPTmS1fExbuA3SnjNx4jLi5Ensc5hRnGNhY9onYw2ok2H\\",2],[\\"EXHRoKfGoCe414hrzYbG8SNCNVgBxSpu6rHUBkPNtZcj3om\\",1]]}}, {\\"callIndex\\":\\"0x0610\\",\\"args\\":{\\"era\\":696,\\"validators\\":[[\\"F3BPTmS1fExbuA3SnjNx4jLi5Ensc5hRnGNhY9onYw2ok2H\\",6],[\\"EXHRoKfGoCe414hrzYbG8SNCNVgBxSpu6rHUBkPNtZcj3om\\",5]]}}, {\\"callIndex\\":\\"0x0610\\",\\"args\\":{\\"era\\":697,\\"validators\\":[[\\"F3BPTmS1fExbuA3SnjNx4jLi5Ensc5hRnGNhY9onYw2ok2H\\",5],[\\"EXHRoKfGoCe414hrzYbG8SNCNVgBxSpu6rHUBkPNtZcj3om\\",4]]}}, {\\"callIndex\\":\\"0x0610\\",\\"args\\":{\\"era\\":698,\\"validators\\":[[\\"F3BPTmS1fExbuA3SnjNx4jLi5Ensc5hRnGNhY9onYw2ok2H\\",6],[\\"EXHRoKfGoCe414hrzYbG8SNCNVgBxSpu6rHUBkPNtZcj3om\\",6]]}}, {\\"callIndex\\":\\"0x0610\\",\\"args\\":{\\"era\\":699,\\"validators\\":[[\\"F3BPTmS1fExbuA3SnjNx4jLi5Ensc5hRnGNhY9onYw2ok2H\\",6],[\\"EXHRoKfGoCe414hrzYbG8SNCNVgBxSpu6rHUBkPNtZcj3om\\",7]]}}, {\\"callIndex\\":\\"0x0610\\",\\"args\\":{\\"era\\":700,\\"validators\\":[[\\"F3BPTmS1fExbuA3SnjNx4jLi5Ensc5hRnGNhY9onYw2ok2H\\",6],[\\"EXHRoKfGoCe414hrzYbG8SNCNVgBxSpu6rHUBkPNtZcj3om\\",7]]}}]",
      "extrinsicIndex": 3,
      "hash": "0x49cc0b81fce3d73dd30b125981ed511addbf121e8bdfed677c77da0d6162b944",
      "isSignedTransaction": true,
      "isSuccess": true,
      "method": "batch",
      "nonce": 42,
      "section": "utility",
      "signature": "0x54cb9be9b44a25b2493512ab9d413c0a6388f6bc4b607693b99d6e5230efc923669e7d4dd621d48dc67301b0380690f9d4994213e0e7d7b88cb3adcafdbe8582",
      "signer": "HA4FRTrk4n5f2gvsfNLYY4b9Tw5eincNBXpxpSV21qGVXvB",
      "time": Object {
        "nanos": 0,
        "seconds": 1587626346,
      },
    },
  ],
}
`;
